packages/itwin/grouping-mapping-widget/src/widget/GroupingMappingWidget.tsx

```diff
@@ -14,19 +14,19 @@ import {
 } from "@itwin/appui-abstract";
 import type { AccessToken } from "@itwin/core-bentley";
 import React from "react";
+import type { ClientPrefix } from "./components/context/GroupingApiConfigContext";
 import GroupingMapping from "./components/GroupingMapping";
-import type { ClientPrefix } from "./components/GroupingMapping";
 import type { IMappingClient } from "./IMappingClient";
 
 export class GroupingMappingProvider implements UiItemsProvider {
   public readonly id = "GroupingMappingProvider";
 
-  private readonly _accessToken?: AccessToken;
+  private readonly _getAccessToken?: () => Promise<AccessToken>;
   private readonly _prefix?: ClientPrefix;
   private readonly _client?: IMappingClient;
 
-  constructor(accessToken?: AccessToken, prefix?: "" | "dev" | "qa", client?: IMappingClient) {
-    this._accessToken = accessToken;
+  constructor(getAccessToken?: () => Promise<AccessToken>, prefix?: "" | "dev" | "qa", client?: IMappingClient) {
```

Comment:
Since you are making a breaking change here, maybe change all these parameters to a single optional params object? It's pretty annoying when I want to provide only the `client` - need to pass 2 `undefined` values before I can pass in the client.

Reply:
Maybe `GroupingMappingProps` could be reused here.

Reply:
Good idea

Reply:
Converted to a single object params.